### PR TITLE
TRA-3842 daemon acceptance tests

### PR DESCRIPTION
**Pull request by RamiBerm**

### CODE REVIEW

File: acceptanceTests/tap_test.go [hunk 1]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
...
+				}
+			})
+
+			if err := tapCmd.Run(); err != nil {
+				t.Errorf("error occured while running the tap command, err: %v", err)
+				return
+			}
+
+			t.Logf("running command: %v", viewCmd.String())
+			if err := viewCmd.Start(); err != nil {
```
Comment (RoyUP9) [inline acceptanceTests/tap_test.go:1017]: do we wanna run the view command before the tappers are ready?

File: acceptanceTests/tap_test.go [hunk 2]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
...
+				break
+			}
+		}
+
+		if !podFound {
+			t.Errorf("unexpected result - expected pod not found, pod namespace: %v, pod name: %v", expectedPod.Namespace, expectedPod.Name)
+			return
+		}
+	}
+}
```
Comment (RoyUP9) [inline acceptanceTests/tap_test.go:1169]: add newline

File: acceptanceTests/tap_test.go [hunk 3]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
...
+	var namespacesCmd []string
+	for _, expectedPod := range expectedPods {
+		namespacesCmd = append(namespacesCmd, "-n", expectedPod.Namespace)
+	}
+	tapCmdArgs = append(tapCmdArgs, namespacesCmd...)
+
+	tapCmd := exec.Command(cliPath, tapCmdArgs...)
+	t.Logf("running command: %v", tapCmd.String())
+
+	viewCmd := exec.Command(cliPath, getDefaultViewCommandArgs()...)
```
Comment (RoyUP9) [inline acceptanceTests/tap_test.go:1103]: same comment here about the view command

File: acceptanceTests/tap_test.go [hunk 4]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
...
+			t.Cleanup(func() {
+				if err := runMizuClean(); err != nil {
+					t.Logf("error running mizu clean: %v", err)
+				}
+				if err := cleanupCommand(viewCmd); err != nil {
+					t.Logf("failed to cleanup view command, err: %v", err)
+				}
+			})
+
+			if err := tapCmd.Run(); err != nil {
```
Comment (nimrod-up9) [inline acceptanceTests/tap_test.go:1011]:
```go
t.Logf("running command: %v", tapCmd.String())
```

or delete 1016.

File: acceptanceTests/tap_test.go (hunk 1 above)
Comment (RamiBerm) [reply acceptanceTests/tap_test.go:1017]: we need view to be able to connect to the api server, you're right about the tappers tho, this might check entries too soon

File: acceptanceTests/tap_test.go [hunk 5]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
 		return
 	}
 }
+
+func TestTapDaemon(t *testing.T) {
```
Comment (nimrod-up9) [inline acceptanceTests/tap_test.go:977]: Give it a more detailed name.
This function doesn't test all of daemon mode capabilities and edge cases.
It only tests that traffic is recorded. For example, this test doesn't compare the recorded traffic to the transmitted messages, and it doesn't check that pods are filtered correctly.
This test is good in itself. I think that the names should be accurate so that we are clear on what the tests cover.
Maybe `TestDaemonSeeTraffic` or `TestDaemonListEntries`?

File: acceptanceTests/tap_test.go [hunk 6]
```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
...
+			}
+			if err := retriesExecute(shortRetriesCount, entriesCheckFunc); err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+		})
+	}
+}
+
+func TestTapDaemonMultipleNamespaces(t *testing.T) {
```
Comment (nimrod-up9) [inline acceptanceTests/tap_test.go:1073]: Also needs a more detailed name.
From just looking at the name of this function, I thought that it tests the same thing as `TestTapDaemon` only in the specified namespaces, but it doesn't.
Maybe `TestDaemonSeePods`?

Comment (nimrod-up9) [reply acceptanceTests/tap_test.go:1073]: Add a negative test (in this function or a different one) that tests that a pod in another namespace is not seen. (`unexpectedPods`)

File: acceptanceTests/tap_test.go (hunk 3 above)
Comment (RamiBerm) [reply acceptanceTests/tap_test.go:1103]: its necessary so the test can connect to the api

File: acceptanceTests/tap_test.go (hunk 5 above)
Comment (RamiBerm) [reply acceptanceTests/tap_test.go:977]: Was just following naming convention, this is a daemon version of the test `TestTap`, didnt realize this would be misleading

Comment (nimrod-up9) [reply acceptanceTests/tap_test.go:977]: I think it is 🤷‍♂️ 
You don't have to fix all of the tests, just the new ones.

**Review by RoyUP9 — APPROVED** [review]

**Review by nimrod-up9 — APPROVED** [review]
